### PR TITLE
Fix corrupt container on startup

### PR DESCRIPTION
**Pull request by pvizeli**
## Proposed change

Help to get HA back running after a reboot or poweroff with corrupt core container.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/homeassistant/core.py [hunk 1]
```diff
@@ -392,7 +397,7 @@ async def _block_till_run(self, version: str) -> None:
                 break
 
         self._error_state = True
-        raise HomeAssistantError()
+        raise HomeAssistantCrashError()
```
Comment (ludeeus) [inline supervisor/homeassistant/core.py:400]: This needs an update on line 187 as well

Comment (ludeeus) [reply supervisor/homeassistant/core.py:400]: Nevermind  `HomeAssistantCrashError(HomeAssistantError)`

File: supervisor/core.py [hunk 2]
```diff
@@ -241,8 +242,16 @@ async def start(self):
...
             ):
-                with suppress(HomeAssistantError):
+                try:
                     await self.sys_homeassistant.core.start()
+                except HomeAssistantCrashError as err:
+                    _LOGGER.warning("Can't start Home Assistant Core - rebuild")
+                    self.sys_capture_exception(err)
+
+                    with suppress(HomeAssistantError):
+                        await self.sys_homeassistant.core.rebuild()
```
Comment (ludeeus) [inline supervisor/core.py:252]: Should we check for space issues before throwing out the existing?

File: supervisor/homeassistant/core.py (hunk 1 above)
Comment (pvizeli) [reply supervisor/homeassistant/core.py:400]: Line should be 187 catch all exception from this function

File: supervisor/core.py (hunk 2 above)
Comment (pvizeli) [reply supervisor/core.py:252]: We don't raise the Crash error if the diskspace is an issue to build the container

File: supervisor/homeassistant/core.py (hunk 1 above)
Comment (ludeeus) [reply supervisor/homeassistant/core.py:400]: Yup, which is why I marked it as resolved :)

File: supervisor/core.py (hunk 2 above)
Comment (ludeeus) [reply supervisor/core.py:252]: I somehow thought that the rebuild pulled a new image.
But it only does a restart

**Review by ludeeus — APPROVED** [review]

**Pull request merged**